vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/data_transformation.tsx

```diff
@@ -0,0 +1,275 @@
+import {some, Maybe, withDefault, isSome} from '../../../../selectors/maybe';
+import {useMemoized} from '../../../../selectors/workflow_selectors';
+
+const useStyles = makeStyles((theme) => ({
+  dialog: {
+    maxWidth: '90vw',
+    maxHeight: '90vh'
+  },
+  subtitle: {display: 'inline'}
+}));
+
+function DataTransformationModal({
+  data,
+  onChange,
+  onClose
+}: {
+  data: any[][];
+  onChange: (data: Maybe<{[key: string]: any}>) => void;
+  onClose: () => void;
+}) {
+  const classes = useStyles();
+
+  const hotTableComponent = useRef<any>(null);
+  const hyperformulaInstance = useMemo(
+    () =>
+      HyperFormula.buildEmpty({
+        // to use an external HyperFormula instance,
+        // initialize it with the `'internal-use-in-handsontable'` license key
+        licenseKey: 'internal-use-in-handsontable'
+      }),
+    []
+  );
+
+  return (
+    <>
+      <DialogTitle>
+        Transform your data (
+        <Typography className={classes.subtitle}>
+          <Link
+            target='_blank'
+            rel='noreferrer'
+            href='https://handsontable.github.io/hyperformula/guide/built-in-functions.html'
+          >
+            Supported functions
+          </Link>
+        </Typography>
+        )
+      </DialogTitle>
+      <DialogContent className={classes.dialog}>
+        <HotTable
+          ref={hotTableComponent}
+          settings={{
+            data: data,
+            colHeaders: true,
+            rowHeaders: true,
+            height: 'auto',
+            licenseKey: 'non-commercial-and-evaluation',
+            formulas: {
+              engine: hyperformulaInstance
+            },
+            contextMenu: {
+              items: {
+                col_left: {
+                  name: 'Insert column to left'
+                },
+                col_right: {
+                  name: 'Insert column to right'
+                },
+                remove_col: {},
+                undo: {},
+                redo: {},
+                clear_column: {}
+              }
+            }
+          }}
+        />
+      </DialogContent>
+      <DialogActions>
+        <Button
+          onClick={onClose}
+          startIcon={<CancelIcon />}
+          color='secondary'
+          variant='contained'
+        >
+          Cancel
+        </Button>
+        <Button
+          onClick={() => {
+            if (hotTableComponent.current) {
+              // hotTableComponent.current.hotInstance.getSourceData()
+              //   returns the raw formulas, unrendered, which we
+              //   set as input values to preserve any transformations.
+              const sourceData = hotTableComponent.current.hotInstance.getSourceData();
+              const data = hotTableComponent.current.hotInstance.getData();
+
+              onChange(
+                some({
+                  source_data: some(nestedArrayToDataFrameJson(sourceData)),
+                  data: some(nestedArrayToDataFrameJson(data))
```

Comment:
Perhaps we can rename these to `formulaic_data` and `calculated_data` or something similar?  I think it'd be nice to have the contents of these both be obvious from the name as they're the outputs that workflow creators will get to work with.

Reply:
Sure, I can do that! `data` and `source_data` are just what the table library uses, so I mimic'd by default. I'll change those to be more descriptive.